temba/channels/models.py

```diff
@@ -1310,30 +1317,43 @@ def send_jasmin_message(cls, channel, msg, text):
 
     @classmethod
     def send_junebug_message(cls, channel, msg, text):
-        from temba.msgs.models import WIRED
+        from temba.msgs.models import WIRED, Msg
 
         # the event url Junebug will relay events to
         event_url = 'https://%s%s' % (
             settings.HOSTNAME,
             reverse('handlers.junebug_handler',
                     args=['event', channel.uuid]))
 
+        is_ussd = channel.channel_type == Channel.TYPE_JUNEBUG_USSD
+
         # build our payload
         payload = dict()
-        payload['to'] = msg.urn_path
         payload['from'] = channel.address
         payload['event_url'] = event_url
         payload['content'] = text
 
+        if is_ussd:
+            external_id, session_status = Msg.objects.values_list('response_to__external_id', 'session__status').get(pk=msg.id)
+            # NOTE: Only one of `to` or `reply_to` may be specified
+            payload['reply_to'] = external_id
+            payload['channel_data'] = {
+                'continue_session': session_status not in ChannelSession.DONE,
```

Comment:
@nullcode I noticed that in #423 you've added a boolean property to session called `should_end` that checks for the `status` being in state `ENDING`. Is this perhaps what I should be using here too? Is `COMPLETED` as a status only set after the message has been sent and am I checking that too early at this point?